debugger/src/main/java/org/codice/acdebugger/breakpoints/SecurityCheckInformation.java

```diff
@@ -133,103 +111,33 @@
-      final String domain =
-          (domainValue == currentDomainValue) ? currentDomain : debug.locations().get(domainValue);
-
-      this.context.add(domain);
-      if (i == local_i) {
-        // we know we don't have privileges since we failed here so continue but only after
-        // having added the current domain to the context list
-        continue;
-      } else if (i < local_i) { // we know we have privileges since we failed after `i`
-        debug.permissions().grant(domain, permissionInfos);
-        privilegedDomains.add(domain);
-      } else if (domainValue instanceof ObjectReference) {
-        if (debug.permissions().implies(domain, permissionInfos)) { // check cache
-          privilegedDomains.add(domain);
-        } else if (debug
-            .permissions()
-            .implies((ObjectReference) domainValue, permission)) { // check attached VM
-          debug.permissions().grant(domain, permissionInfos);
-          privilegedDomains.add(domain);
-        }
-      }
-    }
-    this.stack = new ArrayList<>(debug.thread().frameCount());
-    // don't cache the set of stack as it will change every time we invoke()
-    // something using the thread
-    for (int i = 0; i < debug.thread().frameCount(); i++) {
-      final StackFrame frame = debug.thread().frame(i);
-      final Location location = frame.location(); // cache before we invoke anything on the thread
-      final ObjectReference thisObject = frame.thisObject();
-      final String domain = debug.locations().get(frame);
-      final StackFrameInformation currentFrame =
-          new StackFrameInformation(domain, location, thisObject);
-
-      stack.add(currentFrame);
-    }
-    if (currentDomain == null) {
+    this.debug = debug;
+    this.context = context;
+    this.domains = new ArrayList<>(context.getDomains().size());
+    if (context.getCurrentDomain() == null) {
       // since bundle-0/boot domain always has all permissions, we cannot received null as the
       // current domain where the failure occurred
       dumpTroubleshootingInfo(
-          debug, "AN ERROR OCCURRED WHILE ATTEMPTING TO FIND THE LOCATION FOR A DOMAIN");
-      throw new Error("unable to find location for domain: " + currentDomainValue.type().name());
+          "AN ERROR OCCURRED WHILE ATTEMPTING TO FIND THE LOCATION FOR A DOMAIN");
+      throw new Error(
+          "unable to find location for domain: "
+              + context.getCurrentDomainReference().type().name());
     }
     this.invalid = !recompute();
-    if (!domains.contains(currentDomain)) {
-      // it would seem we are unable to find the domain where we failed after recomputing
-      dumpTroubleshootingInfo(
-          debug,
-          "AN ERROR OCCURRED WHILE ATTEMPTING ANALYZE THE SECURITY EXCEPTION, THE",
-          "DOMAIN WHERE THE FAILURE WAS REPORTED CANNOT BE FOUND FROM THE CURRENT",
-          "ACCESS CONTROL CONTEXT");
-    }
-    analyze0(debug);
+    analyze0();
```

Comment:
ℹ️ The failure tracked on the right side is no longer possible with the re-implemented way to compute combined domains. In addition, there is a check there to detect major errors.